templates/dotnet/src/Appwrite/ID.cs.twig

```diff
@@ -1,15 +1,44 @@
+using System;
+
 namespace {{ spec.title | caseUcfirst }}
 {
     public static class ID
     {
-        public static string Unique()
+        // Generate an hex ID based on timestamp
+        // Recreated from https://www.php.net/manual/en/function.uniqid.php
+        private static string HexTimestamp()
         {
-            return "unique()";
+            var now = DateTime.UtcNow;
+            var epoch = (now - new DateTime(1970, 1, 1));
+            var secondsSinceEpoch = (long)epoch.TotalSeconds;
+            var msecs = epoch.TotalMilliseconds - secondsSinceEpoch * 1000;
+            var microsecondsPart = (long)(msecs * 1000); // Convert milliseconds to microseconds
```

Comment:
Why don't you go directly like:     final msecs = now.microsecondsSinceEpoch - secondsSinceEpoch * 1000000;